sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala

```diff
@@ -113,7 +113,10 @@ class HiveSqlAstBuilder extends SparkSqlAstBuilder {
     if (ctx.REPLICATION != null) {
       logWarning("REPLICATION clause is ignored.")
     }
-    DropTable(visitTableIdentifier(ctx.tableIdentifier).toString, ctx.EXISTS != null)
+    DropTable(
+      visitTableIdentifier(ctx.tableIdentifier),
+      ctx.EXISTS != null,
```

Comment:
@yhuai It is hard to recognize `Trash` dir. The trash feature is disabled by default.

First, users need to enable `Trash`. http://www.cloudera.com/documentation/archive/cdh/4-x/4-7-1/CDH4-Installation-Guide/cdh4ig_topic_11_2.html#topic_11_2_8_unique_1

Second, the path of `Trash` directory depends on many factors. 

> If the trash directory is not accessible the data will remain in the original location 

> The data is moved to the trash directory in the home directory of the user executing the hive command. 

http://www.cloudera.com/documentation/archive/cdh/4-x/4-7-1/CDH4-Installation-Guide/cdh4ig_hive_trash.html